File: app/src/main/java/com/divercity/app/features/home/jobs/JobsViewPagerAdapter.java
package com.divercity.app.features.home.jobs;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import com.divercity.app.R;
import com.divercity.app.features.home.jobs.jobs.JobsListFragment;
import com.divercity.app.features.home.jobs.applications.JobsApplicationsFragment;
import com.divercity.app.features.home.jobs.saved.JobsSavedFragment;

import javax.inject.Inject;

/**
 * Created by lucas on 16/10/2018.
 */

public class JobsViewPagerAdapter extends FragmentStatePagerAdapter {
    private final int PAGE_COUNT = 3;
    // Tab titles
    private String tabTitles[];

    @Inject
    public JobsViewPagerAdapter(Context context, FragmentManager fm) {
        super(fm);
        tabTitles = new String[]{
                context.getString(R.string.jobs),
                context.getString(R.string.applications),
                context.getString(R.string.saved)
        };
    }

    @Override
    public Fragment getItem(int position) {
        Fragment fragment;
        switch (position) {
            case 0:
                fragment = JobsListFragment.newInstance();
                break;
            case 1:
                fragment = JobsApplicationsFragment.newInstance();
                break;
            default:
                fragment = JobsSavedFragment.newInstance();
                break;
        }
        return fragment;
    }

    @Override
    public int getCount() {
        return PAGE_COUNT;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return tabTitles[position];
    }

}

File: app/src/main/java/com/divercity/app/features/home/jobs/JobsViewPagerAdapter.kt
package com.divercity.app.features.home.jobs

import android.content.Context
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter

import com.divercity.app.R
import com.divercity.app.features.home.jobs.applications.JobsApplicationsFragment
import com.divercity.app.features.home.jobs.jobs.JobsListFragment
import com.divercity.app.features.home.jobs.mypostings.MyPostingsFragment
import com.divercity.app.features.home.jobs.saved.SavedJobsFragment
import com.divercity.app.repository.user.UserRepository

import javax.inject.Inject

/**
 * Created by lucas on 16/10/2018.
 */

class JobsViewPagerAdapter
@Inject constructor(
        val context: Context,
        fm: FragmentManager,
        val userRepository: UserRepository
) : FragmentStatePagerAdapter(fm) {

    private val PAGE_COUNT = 3

    // Tab titles
    private val tabTitlesRecruiter: Array<String> = arrayOf(
            context.getString(R.string.applications),
            context.getString(R.string.my_postings),
            context.getString(R.string.jobs)
    )


    // Tab titles
    private val tabTitlesJobSeeker: Array<String> = arrayOf(
            context.getString(R.string.jobs),
            context.getString(R.string.applications),
            context.getString(R.string.saved)
    )

    override fun getItem(position: Int): Fragment {
        return if (userRepository.getAccountType() != null &&
                (userRepository.getAccountType().equals(context.getString(R.string.job_seeker_id)) ||
                        userRepository.getAccountType().equals(context.getString(R.string.student_id)))
        )
            getFragmentsSeeker(position)
        else
            getFragmentsRecruiter(position)
    }

    private fun getFragmentsSeeker(position: Int): Fragment {
        return when (position) {
            0 -> JobsListFragment.newInstance()
            1 -> JobsApplicationsFragment.newInstance()
            else -> SavedJobsFragment.newInstance()
        }
    }

    private fun getFragmentsRecruiter(position: Int): Fragment {
        return when (position) {
            0 -> JobsApplicationsFragment.newInstance()
            1 -> MyPostingsFragment.newInstance()
            else -> JobsListFragment.newInstance()
        }
    }

    override fun getCount(): Int = PAGE_COUNT


    override fun getPageTitle(position: Int): CharSequence? {
        return if (userRepository.getAccountType() != null &&
                (userRepository.getAccountType().equals(context.getString(R.string.job_seeker_id)) ||
                        userRepository.getAccountType().equals(context.getString(R.string.student_id))) ||
                userRepository.getAccountType().equals(context.getString(R.string.professional_id))
        )
            tabTitlesJobSeeker[position]
        else
            tabTitlesRecruiter[position]
    }
}
